### PR TITLE
feat: replace grpcio with tonic

### LINKED ISSUE #240: Try replace `grpcio` with `tonic`
### Describe This Problem

[`grpcio`](https://docs.rs/grpcio/latest/grpcio/) is currently used by CeresDB as gRPC implementation. It's a rust wrapper over the [cpp implementation](https://github.com/grpc/grpc). We have used it for a long time but there are always some issues we are facing:
- long build time https://github.com/CeresDB/ceresdb/issues/151
- memory leak (from C++) https://github.com/grpc/grpc/issues?q=is%3Aissue+leak
- complicated build procedure (from both rust and C++) https://github.com/CeresDB/ceresdb-client-py/issues/20

[`tonic`](https://docs.rs/tonic/latest/tonic/) is a native gRPC implementation with these pros.:
- Native, and safety (maybe? https://github.com/hyperium/tonic/issues?q=is%3Aissue+leak)
- More users (compare to `grpcio`)
- Low memory consumption (see below)

And of cause, its cons.:
- Performance (refer to https://github.com/LesnyRumcajs/grpc_bench/wiki/2022-04-23-bench-results)

  | name                        |   req/s |   avg. latency |        90 % in |        95 % in |        99 % in | avg. cpu |   avg. memory |
  |---|---|---|---|---|---|---|---|
  | rust_tonic_mt               |   64091 |       13.53 ms |       23.42 ms |       28.17 ms |       37.61 ms |  277.41% |     18.16 MiB |
  | rust_grpcio                 |   68377 |       12.11 ms |       18.79 ms |       22.81 ms |       32.61 ms |  259.81% |      25.3 MiB |

  through this table, `grpcio` is faster than `tonic`. QpS on gRPC layer loss around 7%.

### Proposal

With those improvements, I think `tonic` is worth a try

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ShiKaiWi**
# Which issue does this PR close?

Closes #240 

# Rationale for this change
 Refer to the #240.

# What changes are included in this PR?
- Switch the grpc service framework to tonic;
- Upgrade the ceresdbproto:
    - Switch to prost based pb message;
    - Make node heartbeat as a unary call rather than a stream call;
- Remove the grpcio dependency.

# Are there any user-facing changes?
None.

# How does this change test
Unit tests and integration tests.

### CODE REVIEW

File: ceresdbproto_deps/Cargo.toml [hunk 1]
```diff
@@ -9,5 +9,5 @@ workspace = true
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies.ceresdbproto]
-git = "https://github.com/CeresDB/ceresdbproto.git"
-rev = "8e54107e0255429313024822ec85b78561939d00"
\ No newline at end of file
+git = "https://github.com/ShiKaiWi/ceresdbproto.git"
```
Comment (ShiKaiWi) [inline ceresdbproto_deps/Cargo.toml:12]: After https://github.com/CeresDB/ceresdbproto/pull/21 merged, update this repo url and the commit.

File: ceresdbproto_deps/Cargo.toml [hunk 2]
```diff
@@ -10,4 +10,4 @@ workspace = true
 
 [dependencies.ceresdbproto]
 git = "https://github.com/CeresDB/ceresdbproto.git"
-rev = "8e54107e0255429313024822ec85b78561939d00"
\ No newline at end of file
+rev = "ecc410b8ccd79c89391788597d977afe6d6a98ed"
```
Comment (fengjiachun) [inline ceresdbproto_deps/Cargo.toml:13]: Add a new line？

Comment (ShiKaiWi) [reply ceresdbproto_deps/Cargo.toml:13]: Ok.

File: server/Cargo.toml [hunk 3]
```diff
@@ -41,6 +41,8 @@ sql = { path = "../sql" }
 system_catalog = { path = "../system_catalog" }
 table_engine = { path = "../table_engine" }
 tokio = { version = "1.0", features = ["full"] }
+tokio-stream = {version = "0.1", features = ["net"]}
```
Comment (chunshao90) [inline server/Cargo.toml:44]:
```suggestion
tokio-stream = { version = "0.1", features = ["net"] }
```

File: meta_client/Cargo.toml [hunk 4]
```diff
@@ -24,4 +23,5 @@ serde_derive = "1.0.81"
 serde_json = "1.0.60"
 snafu = { version ="0.6.10", features = ["backtraces"]}
```
Comment (chunshao90) [inline meta_client/Cargo.toml:24]:
```suggestion
snafu = { version ="0.6.10", features = ["backtraces"] }
```

**Review by chunshao90 — APPROVED** [review]
LGTM

File: server/Cargo.toml (hunk 3 above)
Comment (ShiKaiWi) [reply server/Cargo.toml:44]: I will fix this in the next PR.

**Pull request merged**